### PR TITLE
Course edits now can be disabled for single course, closes issue #1440

### LINKED ISSUE #1440: Allow course edits to be disabled for a single course
Probably this could be done the same way we handle enabling/disabling timelines, with another entry in the course flags.

See https://github.com/WikiEducationFoundation/WikiEduDashboard/commit/1b66177d705d1836340ca7612b478f570c1c59f2

### PR DESCRIPTION
**Pull request by vzhz**
This PR closes #1440 :
- add front-end toggle for `wiki_edit_enabled`
- adds flag for single-course edits
- refactors `wiki_edit_enabled` into the `Course` class, now removed from subclasses

### CODE REVIEW

**Comment by psinghal20** [comment]
@vzhz Please take a look at the failing Travis build. There are several linting errors, which are causing the build to fail.

File: app/assets/javascripts/components/overview/course_edit_toggle.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import CourseActions from '../../actions/course_actions.js';
+import uuid from 'uuid';
+
+const CourseEditToggle = React.createClass({
```
Comment (ragesoss) [inline app/assets/javascripts/components/overview/course_edit_toggle.jsx]: This looks like it duplicates a lot of the logic from YesNoSelector, which was recently extracted. Could you use that instead? See PrivacySelector, for example.

File: app/controllers/courses_controller.rb [hunk 2]
```diff
@@ -208,6 +208,17 @@ def set_timeline_enabled
     end
   end
 
+  def set_course_edits_enabled
```
Comment (ragesoss) [inline app/controllers/courses_controller.rb]: Doesn't look like this method gets called.

**Comment by ragesoss** [comment]
This should include a test that shows a course that responds 'wiki_edits_enabled?' true initially, then after interacting with the selector and saving it, responds false.

**Comment by vzhz** [comment]
Hey team, I've got the linter errors taken care of (locally, will push it all together) and am about to work on tracking down the uncalled "set_course_edits_enabled" that @ragesoss pointed out doesn't get called. Thanks for the comments, will be working through them today.

File: app/controllers/courses_controller.rb (hunk 2 above)
Comment (vzhz) [reply app/controllers/courses_controller.rb]: Fixed, thank you!

**Comment by ragesoss** [comment]
@vzhz please have a look at the travis build and check out the failing specs.

What I expect from this change is that the current behavior will be unchanged, except for a course that has a flag set to *disable* edits. The flag would just override the case where edits are currently enabled and disable them for that course.

**Comment by vzhz** [comment]
@ragesoss, do I understand that all courses should be editable by default and our flags should override this starting condition? Thanks!

**Comment by ragesoss** [comment]
Thanks for this! I just merged a fix for #1440, and reviewing this PR was useful as I was sorting out the flow.

**Pull request closed without merging**